ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -105,11 +106,11 @@
         offset = new Offset(logSegmentName, value.getLong());
         flags = value.get();
         long expiresAt = value.getInt();
-        expiresAtMs = expiresAt != Utils.Infinite_Time ? (expiresAt * Time.MsPerSec) : Utils.Infinite_Time;
+        expiresAtMs = expiresAt > Utils.Infinite_Time ? (expiresAt * Time.MsPerSec) : Utils.Infinite_Time;
```

Comment:
minor: I think that the check should be `expiresAt >= 0` because, with this check, you are saying here that you know `Utils.Infinite_Time` is negative (or lower than all valid times).